src/posthog-people.ts

```diff
@@ -1,176 +0,0 @@
-/* eslint camelcase: "off" */
-import { addOptOutCheck } from './gdpr-utils'
-import { _base64Encode, _copyAndTruncateStrings, _each, _encodeDates, _extend, _info, _isObject } from './utils'
-import { PostHogConfig, Properties, RequestCallback } from './types'
-import { PostHog } from './posthog-core'
-
-const SET_ACTION = '$set'
-const SET_ONCE_ACTION = '$set_once'
-
-/**
- * PostHog People Object
- * @constructor
- */
-class PostHogPeople {
-    _posthog: PostHog
-
-    set: (prop: string | Properties, to?: string, callback?: RequestCallback) => void
-    set_once: (prop: string | Properties, to?: string, callback?: RequestCallback) => void
-
-    constructor(posthog: PostHog) {
-        this._posthog = posthog
-
-        /*
-         * Set properties on a user record.
-         *
-         * ### Usage:
-         *
-         *     posthog.people.set('gender', 'm');
-         *
-         *     // or set multiple properties at once
-         *     posthog.people.set({
-         *         'Company': 'Acme',
-         *         'Plan': 'Premium',
-         *         'Upgrade date': new Date()
-         *     });
-         *     // properties can be strings, integers, dates, or lists
-         *
-         * @param {Object|String} prop If a string, this is the name of the property. If an object, this is an associative array of names and values.
-         * @param {*} [to] A value to set on the given property name
-         * @param {Function} [callback] If provided, the callback will be called after capturing the event.
-         */
-        this.set = addOptOutCheck(posthog, (prop: string | Properties, to?: string, callback?: RequestCallback) => {
-            const data = this.set_action(prop, to)
-            if (_isObject(prop)) {
-                callback = to as any
-            }
-            // make sure that the referrer info has been updated and saved
-            if (this._get_config('save_referrer')) {
-                this._posthog.persistence.update_referrer_info(document.referrer)
-            }
-
-            // update $set object with default people properties
-            data[SET_ACTION] = _extend(
-                {},
-                _info.people_properties(),
-                this._posthog.persistence.get_referrer_info(),
```

Comment:
This bit (or other places) I could be important 🤷‍♀️  Can we make sure we're sending `$referrer` properties on the event, which provide `$inital_referrer` later in the plugin-server on the person.

Reply:
oh heh, similar names. Can we add `setPersonPropertiesForFlags()` to the set call ? We want this to run whenever set is called (not set_once)

Reply:
I'll see if I can add a test that makes sense for that functionality! I think the test that would fail now no longer exists.

Reply:
@neilkakkar I've added some feature flag tests around this [here](https://github.com/PostHog/posthog-js/pull/584/files#diff-8db8793d500ddffe9c45113068c6a1aa6c87b73ea7563ac70a54d53f9c9891d9R90)

The second is failing however, and I'm not sure why 🤷

Reply:
Yep, I expect because calling identify with the same distinct id is a no-op, it doesn't even send an `$identify` event, just a person property update.

Reply:
Right. We do however still send an `/engage/` request in these cases, I was expecting it to still be triggered there. And in these changes we're sending a $set, so should also still be sent to decide.

I'm going to separate out the new feature flag tests from this PR as it's getting a little big 😱

Reply:
Ah yeah, that makes sense 🤔 - that update should trigger decide requests too.